serve/tests/unittest/test_engine_with_samplers.py

```diff
@@ -206,16 +205,16 @@ def _test_stop(
             assert len(res.sequences) == 1
             seq = res.sequences[0]
             req_id = int(res.request_id)
+            generated[int(res.request_id)] += seq.delta
+
             if seq.is_finished:
                 assert seq.finish_reason == FinishReason.Stop, f"{seq.finish_reason.name}"
-                assert not seq.delta
```

Comment:
@sunggg Due to the change in `sync_engine.py`, we are no longer sending an empty sequence. `seq.is_finished` becomes True when the last token is generated. So this test needed to be updated.